llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -1088,7 +1089,7 @@ class concat_iterator
     if (Begin == End)
       return {};
 
-    if constexpr (ReturnsByValue)
+    if constexpr (ReturnsValueOrPointer)
       return *Begin;
```

Comment:
So this wraps pointers with `std::optional`? Why can't we return pointers here?

Reply:
Fixed :+1:.